File: java-version/src/main/java/com/example/java_version/repository/SampleUserRepository.java
package com.example.java_version.repository;

import com.example.java_version.entity.SampleUser;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface SampleUserRepository extends JpaRepository<SampleUser, Long> {
    Optional<SampleUser> findUserByEmail(String email);
}

File: kotlin-version/src/main/kotlin/com/example/kotlin_version/repository/SampleUserRepository.kt
package com.example.kotlin_version.repository

import com.example.kotlin_version.entity.SampleUser
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface SampleUserRepository: JpaRepository<SampleUser, Long> {
    fun findUserByEmail(email: String): SampleUser?
}